src/main/java/org/broadinstitute/hellbender/tools/spark/sv/SVVariantConsensusCall.java

```diff
@@ -133,15 +133,20 @@ static String produceVariantId(final NovelAdjacencyReferenceLocations novelAdjac
 
     // TODO: 12/14/16 does not work for simple translocation
     @VisibleForTesting
-    static int getSvLength(final NovelAdjacencyReferenceLocations breakpoints) {
+    static int getSvLength(final NovelAdjacencyReferenceLocations breakpoints, GATKSVVCFHeaderLines.SVTYPES svtype) {
         final int start = breakpoints.leftJustifiedLeftRefLoc.getEnd();
         final int end = breakpoints.leftJustifiedRightRefLoc.getStart();
 
-        if (start != end) {
-            return end - start;
-        } else {
-            return breakpoints.complication.insertedSequenceForwardStrandRep.length()
-                    + (breakpoints.complication.dupSeqRepeatNumOnCtg - breakpoints.complication.dupSeqRepeatNumOnRef)*breakpoints.complication.dupSeqForwardStrandRep.length();
+        switch (svtype) {
```

Comment:
Rather than have switch statements on SVTYPES scattered around the code, it might be nice to create abstract methods on the SVType enum and override them when necessary. That way, when we add a new SV type, we don't have to search for all switch statements and make sure they support the new type correctly. Is that a refactoring you'd be comfortable doing as part of this PR? Other places to modify would include the `produceAlleles` and `produceVariantIds` methods.